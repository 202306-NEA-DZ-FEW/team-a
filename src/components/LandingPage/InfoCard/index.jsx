function InfoCard({ title, description, icon, className, bgImage }) {
  return (
    <div
      className={`${className} p-4 flex flex-col flex-1 justify-center items-center gap-4`}
      style={{ backgroundImage: bgImage }}
    >
      {icon}
      <h2 className='text-2xl md:text-3xl font-bold'>{title}</h2>
      <p className='text-md font-light'>{description}</p>
    </div>
  );
}

export default InfoCard;
