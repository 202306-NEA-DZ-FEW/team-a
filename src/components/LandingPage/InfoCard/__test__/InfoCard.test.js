import { GiBallPyramid } from "react-icons/gi";
import renderer from "react-test-renderer";

import InfoCard from "..";

describe("InfoCard Component", () => {
  const props = {
    title: "Test Title",
    description: "Test Description",
    icon: <GiBallPyramid />,
  };
  it("renders correctly with props", () => {
    const component = renderer.create(<InfoCard {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly and match snapshot", () => {
    const component = renderer.create(<InfoCard {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
