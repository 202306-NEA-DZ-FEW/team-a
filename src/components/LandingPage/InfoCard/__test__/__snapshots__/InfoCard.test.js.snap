// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard Component renders correctly and match snapshot 1`] = `
<div
  className="undefined p-4 flex flex-col flex-1 justify-center items-center gap-4"
  style={
    Object {
      "backgroundImage": undefined,
    }
  }
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 50c-21.6 0-39 17.35-39 39 0 21.6 17.4 39 39 39s39-17.4 39-39c0-21.65-17.4-39-39-39zm-48 89c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm-144 90c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm-240 89.4c-21.65 0-39 17.4-39 39s17.35 39 39 39c21.6 0 39-17.4 39-39s-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zM64 409c-21.65 0-39 17.4-39 39s17.35 39 39 39 39-17.4 39-39-17.35-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39z"
    />
  </svg>
  <h2
    className="text-2xl md:text-3xl font-bold"
  >
    Test Title
  </h2>
  <p
    className="text-md font-light"
  >
    Test Description
  </p>
</div>
`;

exports[`InfoCard Component renders correctly with props 1`] = `
<div
  className="undefined p-4 flex flex-col flex-1 justify-center items-center gap-4"
  style={
    Object {
      "backgroundImage": undefined,
    }
  }
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 50c-21.6 0-39 17.35-39 39 0 21.6 17.4 39 39 39s39-17.4 39-39c0-21.65-17.4-39-39-39zm-48 89c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm-144 90c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm-240 89.4c-21.65 0-39 17.4-39 39s17.35 39 39 39c21.6 0 39-17.4 39-39s-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zM64 409c-21.65 0-39 17.4-39 39s17.35 39 39 39 39-17.4 39-39-17.35-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39zm96 0c-21.6 0-39 17.4-39 39s17.4 39 39 39 39-17.4 39-39-17.4-39-39-39z"
    />
  </svg>
  <h2
    className="text-2xl md:text-3xl font-bold"
  >
    Test Title
  </h2>
  <p
    className="text-md font-light"
  >
    Test Description
  </p>
</div>
`;
